h2o-r/h2o-package/R/connection.R

```diff
@@ -844,17 +844,17 @@ h2o.resume <- function(recovery_dir=NULL) {
   if (nzchar(own_jar) && is_url(own_jar)) {
     h2o_url <- own_jar # md5 must have same file name and .md5 suffix
     md5_url <- paste(own_jar, ".md5", sep = "")
+    md5_file <- tempfile(fileext = ".md5")
+    download.file(url = md5_url, destfile = md5_file, mode = "w", cacheOK = FALSE, quiet = TRUE)
+    md5_check <- readLines(md5_file, n = 1L)
+    unlink(md5_file)
   } else {
     base_url <- paste("s3.amazonaws.com/h2o-release/h2o", branch, version, "Rjar", sep = "/")
     h2o_url <- paste("https:/", base_url, "h2o.jar", sep = "/")
-    # Get MD5 checksum
-    md5_url <- paste("https:/", base_url, "h2o.jar.md5", sep = "/")
+    md5_check <- "CURRENT_R_PACKAGE_MD5_CHECKSUM"
```

Comment:
Hi @tomasfryda, please can you explain us the execution path of this else brach? why md5_check is set to a string constant?

Reply:
So instead of download it each time, we save the checksum in the release process and after compare the saved one with the one we got from the another downloaded jar.. it make sence more the the TRUE branch

`if (nzchar(own_jar) && is_url(own_jar)) {`

if I am right, we download a file here... download the same file after 

`cat("(This could take a few minutes, please be patient...)\n")`

and compare the checksums... is the key difference here the different mode?

`mode = "w"` vs `mode = "wb"`

Reply:
Yes, thank you! All good for me :)